apps/ngrx-1/src/app/app.component.ts

```diff
@@ -60,14 +50,11 @@ import { loadUsers } from './store/user/user.actions';
 export class AppComponent implements OnInit {
   private store = inject(Store);
 
-  activities$ = this.store.select(selectActivities);
+  activities$ = this.store.select(selectVm);
 
   ngOnInit(): void {
-    this.store.dispatch(loadActivities());
-    this.store.dispatch(loadUsers());
-    this.store.dispatch(loadStatuses());
+    this.store.dispatch(appLoaded());
   }
-
-  getAllTeachersForActivityType$ = (type: ActivityType) =>
-    this.store.select(selectAllTeachersByActivityType(type));
 }
+
+export const appLoaded = createAction('[AppComponent] Loaded');
```

Comment:
maybe in a different file might be better. 

if you are using Nx, in a seperate lib is better also because if the action is called at multiple place, you will get circular dependancies. (I should make a challenge about this as well, might be interesting)

Reply:
What would you name the file in this case? This one is sort of a special case where it's about if the app loaded.

But what if you had an action that represents when a particular page loaded? e.g. `UsersComponent`. Should that action go in the `UsersComponent` like what I've done here? Or should it go into its own file? What should that file called?

Reply:
I name it the same way as the component that dispatch it.
`app.actions.ts`

So if you create another appLoaded action inside another component. it will be inside `users.actions.ts`

What I like to do as well it, create a barrel object 

```ts
const appLoaded = createAction('[AppComponent] Loaded') // no export there

export const AppActions = { appLoaded} 
```

that way, it's very easy to see where the action is coming from when you use it: `store.dispatch(AppActions.appLoaded)`

It's a bit similar to use createActionGroup

Reply:
Love it. Will make the change.

Would you do the same for selectors that are core to a component? e.g. like my `vm` selector approach. Maybe something like `app.selectors.ts`?

Reply:
@tomalaforge sorry to bring up this old thread.

I've been thinking about the separation of actions and reducers/effects in NgRx.

So I understand it's generally advised to have actions as `events` instead of `commands`. Such as `[AppComponent] loaded` instead of `[Users] Load Users`.

But going with the `commands` approach, that means your stores, such as the user store, needs to hold references to actions outside of it. Such as the user store listening to `[AppComponent] loaded`.

This goes against the old saying of Separation of Concerns etc (I think). But is this what NgRx generally recommends?

The only time I can see this being a problem is with feature modules that shouldn't be importing anything from core.